Sources/MeiliSearch/Model/Update.swift

```diff
@@ -49,4 +49,43 @@ public struct Update: Codable, Equatable {
 
     }
 
+    public enum Status: Codable, Equatable {
+
+        case enqueued
+        case processed
+        case failed
+
+        public enum CodingError: Error {
+            case unknownStatus
+        }
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            let rawStatus = try container.decode(String.self)
+            switch rawStatus {
+            case "enqueued":
+                self = .enqueued
+            case "processed":
+                self = .processed
+            case "failed":
+                self = .failed
+            default:
+                throw CodingError.unknownStatus
+            }
+        }
+
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.singleValueContainer()
+            switch self {
+            case .enqueued:
+                try container.encode("enqueued")
+            case .processed:
+                try container.encode("processed")
+            case .failed:
+                try container.encode("failed")
+            }
+        }
```

Comment:
Is this method useful for the user? It is used in testing, but other than that I dont see the usage.

Reply:
This might not be but since `Status` is a value from `Update.Result` and `Update` that are `Codable`, swift forces it to be Codable too. If I set it as `Decodable`, I get this error:

![Screenshot 2020-10-14 at 00 36 53](https://user-images.githubusercontent.com/5521682/95927013-967dca80-0db5-11eb-9690-e6c8bacd6284.png)

I removed the implementation.